client/src/js/base/__tests__/ProgressBar.test.js

```diff
@@ -1,233 +1,96 @@
-
-        expect(wrapper.find("div").length).toEqual(3);
-        expect(wrapper.is("div")).toBe(true);
-        expect(wrapper.childAt(0).is("div")).toBe(true);
-        expect(wrapper.find("#test-child").exists()).toBe(true);
+    it("should render when [this.state.fill!=0]", () => {
+        const wrapper = shallow(<AutoProgressBar {...props} />);
+        wrapper.setState({ ...state, fill: 1 });
         expect(wrapper).toMatchSnapshot();
     });
 
-    it("adds custom eventListener in componentDidUpdate when there is a change in 'now' props", () => {
-        const spyCDU = sinon.spy(ProgressBar.prototype, "componentDidUpdate");
-        wrapper = mount(<ProgressBar bsStyle="primary" now={50} />);
-        const spyAddListener = sinon.spy(
-            wrapper
-                .find("div")
-                .at(1)
-                .instance(),
-            "addEventListener"
-        );
-
-        expect(spyCDU.called).toBe(false);
-        expect(spyAddListener.called).toBe(false);
-
-        // No change in now props; event listener should not be added
-        wrapper.setProps({ bsStyle: "danger", now: 50 });
-
-        expect(spyAddListener.called).toBe(false);
-
-        // Change in now props; add event listener
-        wrapper.setProps({ bsStyle: "warning", now: 100 });
+    it("componentDidUpdate() should update this.interval when [this.state.fill !== prevState.fill] and [this.state.fill === 10]", () => {
+        const wrapper = shallow(<AutoProgressBar {...props} />);
+        wrapper.setState({ ...state, fill: 10 });
+        wrapper.setProps({ ...props, interval: 10000 });
+        expect(wrapper.instance().interval).toEqual(9);
+    });
 
-        expect(spyAddListener.calledOnce).toBe(true);
+    it("componentDidUpdate() should call window.clearInterval when [this.props.active=false && prevState.active=true]", () => {
+        const wrapper = shallow(<AutoProgressBar {...props} />);
+        wrapper.instance().interval = 1;
+        window.clearInterval = jest.fn();
+        wrapper.setProps({ props, active: false });
+        expect(window.clearInterval).toHaveBeenCalledWith(1);
+    });
 
-        spyCDU.restore();
-        spyAddListener.restore();
+    it("componentWillUnmount() should call window.clearInterval when component unmount", () => {
+        const wrapper = mount(<AutoProgressBar {...props} />);
+        wrapper.instance().interval = 1;
+        window.clearInterval = jest.fn();
+        wrapper.unmount();
+        expect(window.clearInterval).toHaveBeenCalledWith(1);
     });
+});
 
-    it("calls onTransitionend() when props.now changes", () => {
-        // Calls callback onMoved function if it exists
-        props = {
-            onMoved: jest.fn()
-        };
-        wrapper = mount(<ProgressBar {...props} />);
-        wrapper.instance().onTransitionend();
-        expect(props.onMoved).toHaveBeenCalled();
+describe("<ProgressBar />", () => {
+    let props;
+    let state;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['state' is defined but never used.](https://app.codacy.com/manual/virtool/virtool/pullRequest?prid=4770288)